File: 02890.py
R, C = map(int, input().split())

results = []
for i in range(R):
    lane = input()

    for j in range(len(lane)):
        if lane[j].isdigit():
            num = int(lane[j])
            results.append([num, j])

            break
results = sorted(results, key=lambda x: -x[1])

prints = []
rank = 0; loc = C + 1
for i in range(len(results)):
    if results[i][1] == loc:
        prints.append([results[i][0], rank])
    else:
        rank += 1; loc = results[i][1]
        prints.append([results[i][0], rank])
prints = sorted(prints, key=lambda x: x[0])

for p in prints:
    print(p[1])

File: 02890.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

bool compare(pair<int, int> p1, pair<int, int> p2) {
	return p1.second > p2.second;
}

int main() {
	int R, C; cin >> R >> C;

	vector<pair<int, int>> results;	// 팀 넘버, 배 위치
	for (int i = 0; i < R; i++) {
		string lane; cin >> lane;

		for (int j = 0; j < lane.length(); j++) {
			if (isdigit(lane[j])) {
				int num = lane[j] - '0';
				results.push_back(pair<int, int>(num, j));

				break;
			}
		}
	}
	sort(results.begin(), results.end(), compare);

	vector<pair<int, int>> prints;	// 팀 넘버, 등수
	int rank = 0, loc = C + 1;
	for (int i = 0; i < results.size(); i++) {
		if (results[i].second == loc)
			prints.push_back(pair<int, int>(results[i].first, rank));
		else {
			rank++; loc = results[i].second;
			prints.push_back(pair<int, int>(results[i].first, rank));
		}
	}
	sort(prints.begin(), prints.end());

	for (int i = 0; i < prints.size(); i++)
		cout << prints[i].second << endl;
}
